src/lib/Translation/Extractor/NotificationTranslationExtractor.php

```diff
@@ -0,0 +1,208 @@
+{
+    /** @var \JMS\TranslationBundle\Translation\FileSourceFactory */
+    private $fileSourceFactory;
+
+    /** @var \PhpParser\NodeTraverser */
+    private $traverser;
+
+    /** @var \JMS\TranslationBundle\Model\MessageCatalogue */
+    private $catalogue;
+
+    /** @var \SplFileInfo */
+    private $file;
+
+    /** @var \Doctrine\Common\Annotations\DocParser */
+    private $docParser;
+
+    /** @var \Psr\Log\LoggerInterface */
+    private $logger;
+
+    /** @var \PhpParser\Node */
+    private $previousNode;
+
+    /**
+     * Methods and "domain" parameter offset to extract from PHP code.
+     *
+     * @var array method => position of the "domain" parameter
+     */
+    protected $methodsToExtractFrom = [
+        'success' => 2,
+        'info' => 2,
+        'warning' => 2,
+        'error' => 2,
+    ];
+
+    public function __construct(DocParser $docParser, FileSourceFactory $fileSourceFactory)
+    {
+        $this->docParser = $docParser;
+        $this->fileSourceFactory = $fileSourceFactory;
+        $this->traverser = new NodeTraverser();
+        $this->traverser->addVisitor($this);
+    }
+
+    public function setLogger(LoggerInterface $logger)
+    {
+        $this->logger = $logger;
+    }
+
+    public function enterNode(Node $node)
+    {
+        $methodCallNodeName = null;
+        if ($node instanceof Node\Expr\MethodCall) {
+            $methodCallNodeName = $node->name instanceof Node\Identifier ? $node->name->name : $node->name;
+        }
+
+        if (!is_string($methodCallNodeName)
+            || !in_array(strtolower($methodCallNodeName), array_map('strtolower', array_keys($this->methodsToExtractFrom)))) {
+            $this->previousNode = $node;
+
+            return;
+        }
+
+        $ignore = false;
+        $desc = $meaning = null;
+
+        if (null !== $docComment = $this->getDocCommentForNode($node)) {
+            if ($docComment instanceof Doc) {
+                $docComment = $docComment->getText();
+            }
+
+            foreach ($this->docParser->parse($docComment, 'file ' . $this->file . ' near line ' . $node->getLine()) as $annot) {
+                if ($annot instanceof Ignore) {
+                    $ignore = true;
+                } elseif ($annot instanceof Desc) {
+                    $desc = $annot->text;
+                } elseif ($annot instanceof Meaning) {
+                    $meaning = $annot->text;
+                }
+            }
+        } else {
+            return;
+        }
+
+        if (!$node->args[0]->value instanceof String_) {
+            if ($ignore) {
+                return;
+            }
+
+            $message = sprintf('Can only extract the translation id from a scalar string, but got "%s". Please refactor your code to make it extractable, or add the doc comment /** @Ignore */ to this code element (in %s on line %d).', get_class($node->args[0]->value), $this->file, $node->args[0]->value->getLine());
+
+            if ($this->logger) {
```

Comment:
This check is not needed  `$this->logger` is initialized with instance of `\Psr\Log\NullLogger`